### ISSUE TITLE
[Bug]: Our Delete Modals Need Accessibility

### ISSUE THREAD

**Issue by andrewballantyne** [opendatahub-io/odh-dashboard#1387]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When you're in a delete modal (in fact most of our modals, but let's target delete first since it's an underlying shared component) you should be able to hit enter to confirm the modal.

Accessibility is lacking here, but it should come fairly free by HTML & PF.

### Expected Behavior

Implement a Form around the body of the `DeleteModal`, onSubmit needs to stop the form page refresh (eg. `e.preventDefault`)... You may need to manually call the submit method since the submit button may not have the form context.

### Steps To Reproduce

1. Create a DS Project
2. Enter the DeleteModal for the DS Project
3. Type in the value needed
4. Hit enter

### Workaround (if any)

Manually click on the button

### Open Data Hub Version

```yml
Dashboard: v2.11.0
```

**Comment by lugi0** [comment]
Tracking this downstream as well: https://issues.redhat.com/browse/RHODS-9386
